### ISSUE TITLE
scala_proto_library gets incorrectly stamped with proto_library label

### ISSUE THREAD

**Issue by liucijus** [bazelbuild/rules_scala#1234]
`scala_proto_library` gets incorrectly stamped with `proto_library` label. With strict deps it becomes an invalid `buildozer` command:

```
You can use the following buildozer command:
buildozer 'add deps @foo//proto:proto' //bar:bar
```

Given such targets:
```
# proto/BUILD
proto_library(
    name = "proto",
    srcs = ["foo.proto"],
)

scala_proto_library(
    name = "scala_proto",
    deps = [":proto"],
)
```

Buildozer command should suggest `@foo//proto:scala_proto`

**Comment by liucijus** [comment]
The same issue with `java_proto_library` https://github.com/bazelbuild/bazel/issues/4990

**Comment by liucijus** [comment]
The root cause of incorrect stamping is that proto compilation happens in the aspect context on proto deps, where `scala_proto_library` label is not available (which makes sense, as there may be multiple `scala_proto_library` depending on the same `proto_library` target). 
Possible ways (though none of them are without significant compromises) to mitigate stamping problem:
- move compilation from aspects to `scala_proto_library` - this means every `scala_proto_library` will be fully generating and  rebuilding all proto dependecies.
- stamp according `injecting_rule_kind` and generate labels according to [convention (described under "code organization")](https://docs.bazel.build/versions/master/be/protocol-buffer.html#proto_library). This means that aspect can rewrite their label from `foo_proto` to `foo_scala_proto` under the same package. This idea can be stretched even further by refactoring into phases and adding a phase to allow users to have their own labeling convention.

I'm leaning towards conventional labeling option. @simuons, @ittaiz WDYT?

**Comment by ittaiz** [comment]
Conventional means the first option?
Because that sounds too problematic IMHO performance wise

**Comment by liucijus** [comment]
> Conventional means the first option?
> Because that sounds too problematic IMHO performance wise

No, the second one - which would assume users follow recommended code organization described in the Bazel docs

**Comment by ittaiz** [comment]
SGTM

**Comment by simuons** [comment]
I'd vote for second option (conventional labeling option)

**Comment by liucijus** [comment]
Solved with #1243
